object-cache.php

```diff
@@ -419,15 +466,18 @@ function replace( $id, $data, $group = 'default', $expire = 0 ) {
 	function set( $id, $data, $group = 'default', $expire = 0 ) {
 		$key = $this->key( $id, $group );
 
-		if ( isset( $this->cache[ $key ] ) && ( 'checkthedatabaseplease' === $this->cache[ $key ] ) ) {
+		if ( isset( $this->cache[ $key ] ) && ( 'checkthedatabaseplease' === $this->cache[ $key ][ 'value' ] ) ) {
 			return false;
 		}
 
 		if ( is_object( $data ) ) {
 			$data = clone $data;
 		}
 
-		$this->cache[ $key ] = $data;
+		$this->cache[ $key ] = [
+			'value' => $data,
+			'found' => false, // Set to false as not technically found in memcache at this point.
```

Comment:
@nickdaugherty are you able to confirm if this should be true or false at this point?

Reply:
Thinking about this, I'm wondering if we set to false here (in case someone is using `no_mc_groups` but then if the set is successful further down, we update the found in the local cache to be true? WDYT?

Reply:
> but then if the set is successful further down, we update the found in the local cache to be true

This sounds like the way to go to me.

Reply:
Done in latest commit.